deploy/sre-prometheus/100-monitoring-operatordown.PrometheusRule.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: sre-monitoring-operatordown
+    role: alert-rules
+  name: sre-monitoring-operatordown
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-monitoring-operatordown
+    rules:
+    - alert: ClusterMonitoringOperatorDownSRE
+      # Specifically for the Cluster Monitoring Operator, we want to see if the sum cluster_operator_up (which will turn to 0 when
+      # the operator is DOWN) minus the count of prometheus pods outside of the openshift-* namespaces equals -1. This means that if
+      # the cluster operator is down and there are no rogue prometheus instances this will fire.  If the monitoring operator goes
+      # down but there also is rogue prometheus instances, this would return 0 because of the `absent` and therefore will not fire.
+      expr: |
+        (sum(cluster_operator_up{name="monitoring"}) - absent(count(kube_pod_labels{label_app_kubernetes_io_component="prometheus",namespace!~"openshift-.*"}))) == -1
+      for: 10m
+      labels:
+        severity: critical
+        namespace: openshift-cluster-version
+      annotations:
+        message: The monitoring operator may be down or disabled, and the components it manages may be unavailable or degraded. Cluster upgrades may not complete. For more information refer to 'oc get -o yaml clusteroperator monitoring'.
+    - alert: MultiplePrometheusOperatorsSRE
+      # This alert will fire when there are multiple prometheus operators on the cluster unless the cluster monitoring operator is not down.
+      # We first get the labels from the pod with the namespace, adding a custom label to compare on against the count of the cluster monitoring
+      # operator pods available. If the cluster monitoring operator is up, this won't fire for just having additional prometheus instances. If
+      # the customer has multiple prometheus instances, and the cluster monitoring operator is down, this will fire.
+      expr: |
+         count(label_replace(kube_pod_labels{label_app_kubernetes_io_component="prometheus",namespace!~"openshift-.*"}, "sre_metric", "openshift-sre", "", "(.*)")) by (namespace, sre_metric) unless ignoring (namespace) count(label_replace(cluster_operator_up{name="monitoring"}, "sre_metric", "openshift-sre", "", "(.*)") > 0) by (sre_metric)
```

Comment:
Open question, because I really don't know the answer - how confident are we that this identifies the presence of a prometheus __operator__ and not just another Prometheus? In the past I've spotted customer prometheuses running (due to running istio) that don't have an operator installed, which seems okay. Admittedly I can't remember any of those cluster names to check if they would match this metric.

Ultimately I just harbor a little concern about the potential of accidentally masking genuine cluster-monitoring-operator issues due to the presence of an innocuous Prometheus that's not causing any trouble.

Reply:
This is a good callout, but the main reason as to why I wanted to specifically target only clusterOperatorDown with it. I wish there was a way to see how many of these in the past we've seen the ClusterMonitoringOperator go down but it's not due to the 2nd prometheus operator/instance when it's installed.  

This, though, is one of the main reasons that I am not also creating the OAO PR at the moment.  One of the goals of this is to capture this through a separate alert, so that we can start tracking some data through the TOIL reviews of "ok, this is firing but it had nothing to do with the second prometheus instance" which we would bake into the SOP itself.

Reply:
As for your original question, though, this doesn't detect a prometheus operator at all, just the prometheus pod, which is what we're detecting for at the bottom of the SOP.

In my testing I didn't even see a second prometheus operator interfering directly, as I was able to run a second one for about 30-45 minutes before being able to circle back to my test cluster and everything was still running along fine.

So part of this alert is data collection, part of the alert is 🤞🏼 that we can just send a SL and never be alerted for this again.